keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -204,6 +205,31 @@ class SqlVerificationRepository(
     }
   }
 
+  /**
+   * JOOQ-ified access to MySQL's `json_set` function.
+   *
+   * Updates [field] with [values] retaining any existing entries that are not present in [values].
+   *
+   * @see https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-set
+   */
+  @Suppress("UNCHECKED_CAST")
+  private fun <T> jsonSet(field: Field<T>, values: Map<String, Any?>): Field<T> {
+    require(values.isNotEmpty()) {
+      "MySQL's JSON_SET function cannot be used without values"
+    }
+    return function(
+      "json_set",
+      Map::class.java,
+      field,
+      *values.flatMap {
+        listOf(
+          value("\$.${it.key}"),
+          value(it.value)
+        )
+      }.toTypedArray()
+    ) as Field<T>
```

Comment:
This calls MySQL's [`json_set`](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-set) function which expects first the field containing the JSON blob to merge stuff into, then a vararg of JSON pointers & values to insert.

Reply:
When we initially launch a task via Orca for verification we store the task id in that `metadata` field. Later if someone overrides a failed run I would like to add the user and comment to `metadata` without losing that task id.